src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs

```diff
@@ -0,0 +1,708 @@
+        {
+            string expectedJson;
+            string testJson;
+
+            if (quoted)
+            {
+                string constantAsJson = $@"""{constantAsString}""";
+                expectedJson = @$"{{""FloatNumber"":{constantAsJson},""DoubleNumber"":{constantAsJson}}}";
+
+                string testStringAsJson = $@"""{testString}""";
+                testJson = @$"{{""FloatNumber"":{testStringAsJson},""DoubleNumber"":{testStringAsJson}}}";
+            }
+            else
+            {
+                expectedJson = null;
+
+                string testStringAsJson = $@"{testString}";
+                testJson = @$"{{""FloatNumber"":{testStringAsJson},""DoubleNumber"":{testStringAsJson}}}";
+            }
+
+            StructWithNumbers obj;
+            switch (constantAsString)
+            {
+                case "NaN":
+                    obj = JsonSerializer.Deserialize<StructWithNumbers>(testJson, s_optionsAllowFloatConstants);
+                    Assert.Equal(float.NaN, obj.FloatNumber);
+                    Assert.Equal(double.NaN, obj.DoubleNumber);
+                    break;
+                case "Infinity":
+                    obj = JsonSerializer.Deserialize<StructWithNumbers>(testJson, s_optionsAllowFloatConstants);
+                    Assert.Equal(float.PositiveInfinity, obj.FloatNumber);
+                    Assert.Equal(double.PositiveInfinity, obj.DoubleNumber);
+                    break;
+                case "-Infinity":
+                    obj = JsonSerializer.Deserialize<StructWithNumbers>(testJson, s_optionsAllowFloatConstants);
+                    Assert.Equal(float.NegativeInfinity, obj.FloatNumber);
+                    Assert.Equal(double.NegativeInfinity, obj.DoubleNumber);
+                    break;
+                default:
+                    Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<StructWithNumbers>(testJson, s_optionsAllowFloatConstants));
+                    return;
+            }
+
+            JsonTestHelper.AssertJsonEqual(expectedJson, JsonSerializer.Serialize(obj, s_optionsAllowFloatConstants));
+        }
+
+        private struct StructWithNumbers
+        {
+            public float FloatNumber { get; set; }
+            public double DoubleNumber { get; set; }
+        }
+
+        [Fact]
+        public static void ReadFromString_AllowFloatingPoint()
+        {
+            string json = @"{""IntNumber"":""1"",""FloatNumber"":""NaN""}";
+            ClassWithNumbers obj = JsonSerializer.Deserialize<ClassWithNumbers>(json, s_optionReadFromStrAllowFloatConstants);
+
+            Assert.Equal(1, obj.IntNumber);
+            Assert.Equal(float.NaN, obj.FloatNumber);
+
+            JsonTestHelper.AssertJsonEqual(@"{""IntNumber"":1,""FloatNumber"":""NaN""}", JsonSerializer.Serialize(obj, s_optionReadFromStrAllowFloatConstants));
+        }
```

Comment:
What is this testing that is not being tested in the previous method?

Reply:
This is testing a different combination of flag values.